metricbeat/module/kibana/stats/stats.go

```
@@ -101,17 +131,35 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
 func (m *MetricSet) Fetch(r mb.ReporterV2) {
-	content, err := m.http.FetchContent()
+	intervalMs := m.Module().Config().Period.Nanoseconds() / 1000 / 1000
+	now := time.Now()
+
+	m.fetchStats(r, intervalMs, now)
+	if m.xPackEnabled {
+		m.fetchSettings(r, intervalMs, now)
+	}
+}
+
+func (m *MetricSet) fetchStats(r mb.ReporterV2, intervalMs int64, now time.Time) {
+	content, err := m.statsHTTP.FetchContent()
 	if err != nil {
 		r.Error(err)
 		return
 	}
 
 	if m.xPackEnabled {
-		intervalMs := m.Module().Config().Period.Nanoseconds() / 1000 / 1000
-		eventMappingXPack(r, intervalMs, content)
+		eventMappingStatsXPack(r, intervalMs, now, content)
 	} else {
 		eventMapping(r, content)
 	}
+}
+
+func (m *MetricSet) fetchSettings(r mb.ReporterV2, intervalMs int64, now time.Time) {
+	content, err := m.settingsHTTP.FetchContent()
+	if err != nil {
+		r.Error(err)
```

Should we report an error? I think we will not make use of it.